docs/Validation.md

```diff
@@ -259,6 +259,85 @@ export const ProductEdit = () => (
 
 **Tip**: The custom validator function can return a promise, e.g. to use server-side validation. See next section for details.
 
+## Per Input Validation: useUnique
+
+Validating the uniqueness of a field is a common requirement so React-admin provides the `useUnique` hook that returns a validator for this use case.
+
+It will call the `dataProvider.getList` method with a filter to check whether a record exists with the current value of the input for the field matching the input source.
+
+```js
+const UserCreateForm = () => {
+    const unique = useUnique();
+    return (
+        <SimpleForm>
+            <TextInput source="username" validate={unique()} />
```

Comment:
What if the source is a path (e.g. `user.name`)? Does this work? In any case, you should document it.